services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/ovs/DataPlaneServiceImpl.java

```diff
@@ -78,37 +92,47 @@ private VpcEntity getVpcEntity(NetworkConfiguration networkConfig, String vpcId)
         return result;
     }
 
-    private void buildVpcState(NetworkConfiguration networkConfig, GoalState.Builder goalStateBuilder) throws Exception {
-        List<PortState> portStates = goalStateBuilder.getPortStatesList();
+    private void buildVpcStates(NetworkConfiguration networkConfig, UnicastGoalState unicastGoalState) throws Exception {
+        List<PortState> portStates = unicastGoalState.getGoalStateBuilder().getPortStatesList();
         if (portStates == null || portStates.size() == 0) {
             return;
         }
 
         for (PortState portState: portStates) {
             VpcEntity vpcEntity = getVpcEntity(networkConfig, portState.getConfiguration().getVpcId());
             VpcConfiguration.Builder vpcConfigBuilder = VpcConfiguration.newBuilder();
+            vpcConfigBuilder.setFormatVersion(FORMAT_REVISION_NUMBER);
+            vpcConfigBuilder.setRevisionNumber(FORMAT_REVISION_NUMBER);
             vpcConfigBuilder.setId(vpcEntity.getId());
             vpcConfigBuilder.setProjectId(vpcEntity.getProjectId());
-            vpcConfigBuilder.setName(vpcEntity.getName());
-            vpcConfigBuilder.setCidr(vpcEntity.getCidr());
-            vpcConfigBuilder.setTunnelId(Long.parseLong(vpcEntity.getSegmentationId() + ""));
-
-            networkConfig.getSubnets().stream()
-                    .filter(s -> s.getVpcId().equals(vpcEntity.getId()))
-                    .map(InternalSubnetEntity::getId)
-                    .forEach(id -> {
-                        SubnetId.Builder subnetIdBuilder = SubnetId.newBuilder();
-                        subnetIdBuilder.setId(id);
-                        vpcConfigBuilder.addSubnetIds(subnetIdBuilder.build());
-                    });
 
+            if (vpcEntity.getName() != null) {
+                vpcConfigBuilder.setName(vpcEntity.getName());
+            }
+
+            if (vpcEntity.getCidr() != null) {
+                vpcConfigBuilder.setCidr(vpcEntity.getCidr());
+            }
+
+            //vpcConfigBuilder.setTunnelId();
+
+            if (networkConfig.getSubnets() != null) {
+                networkConfig.getSubnets().stream()
+                        .filter(s -> s.getVpcId().equals(vpcEntity.getId()))
+                        .map(InternalSubnetEntity::getId)
+                        .forEach(id -> {
+                            SubnetId.Builder subnetIdBuilder = SubnetId.newBuilder();
+                            subnetIdBuilder.setId(id);
+                            vpcConfigBuilder.addSubnetIds(subnetIdBuilder.build());
+                        });
```

Comment:
I think that if stream is used properly, it can bring the benefits of performance improvement and code simplicity, so I recommend using stream when necessary.